TELCODOCS#2122: Configuring a local arbiter node (#92314)
openshift/openshift-docs: modules/ipi-install-config-local-arbiter-node.adoc

// Module included in the following assemblies:
//
//  *installing/installing_bare_metal/ipi/ipi-install-installation-workflow.adoc

> :_mod-docs-content-type: CONCEPT

Reviewer: ```suggestion
:_mod-docs-content-type: PROCEDURE
```

Author: Updated as per the suggestion.

[id="ipi-install-config-local-arbiter-node_{context}"]
= Configuring a local arbiter node

> You can configure an {product-title} cluster with two control plane nodes and one local arbiter node to retain high availability (HA) while reducing infrastructure costs. This configuration is supported only for bare-metal installations.

Author: Updated as per the suggestion.

> A local arbiter node is a lower-cost, co-located machine that participates in control plane quorum decisions. Unlike a standard control plane node, the arbiter node does not run the full set of control plane services. This setup allows you to maintain HA in your cluster with only two fully provisioned control plane nodes instead of three.

Reviewer: ```suggestion
A local arbiter node is a lower-cost, co-located machine that participates in control plane quorum decisions. Unlike a standard control plane node, the arbiter node does not run the full set of control plane services. You can use this configuration to maintain HA in your cluster with only two fully provisioned control plane nodes instead of three.
```

Author: Updated as per the suggestion.

[IMPORTANT]
====
You can configure a local arbiter node only. Remote arbiter nodes are not supported.
====

To deploy a cluster with two control plane nodes and one local arbiter node, you must define the following nodes in the `install-config.yaml` file:

* 2 control plane nodes
* 1 arbiter node

> * 0 or more worker nodes, depending on your workload requirements

Reviewer: ```suggestion
* 0 or more compute nodes, depending on your workload requirements
```

Author: Hi @eggfoobar 
Could you please confirm whether we should refer to these as compute nodes or worker nodes?
Thanks!

Author: Updated as per the suggestion.

* 2 control plane nodes

> * 1 arbiter node

Reviewer 2: I think we should add the minimum system requirements for the control plane and arbiter nodes, esp for the arbiter node! Afaik, this is currently 2 Thread / 8 GB RAM / 120GB SSD like disk. 
@eggfoobar @Neilhamza please confirm!

Author: Updated as per the suggestion.
Added the following info:

"The arbiter node must meet the following minimum system requirements:

* 2 vCPU (or threads)
* 8 GiB RAM
* 120 GiB SSD or equivalent storage

You must also ensure that the arbiter node is located in a network environment with end-to-end latency of less than 500 milliseconds, including disk I/O. In high-latency environments, you might need to apply the `etcd` slow profile."

include::snippets/technology-preview.adoc[]

> You must enable the `TechPreviewNoUpgrade` feature set in the `FeatureGate` custom resource (CR) to enable the `imageStreamImportMode` feature. 

Author: Updated as per the suggestion.

> You must enable the `TechPreviewNoUpgrade` feature set in the `FeatureGate` custom resource (CR) to enable the `imageStreamImportMode` feature. 

Author: My bad!
Updated it as follows:
"You must enable the `TechPreviewNoUpgrade` feature set in the `FeatureGate` custom resource (CR) to enable the arbiter node feature."

For more information about feature gates, see "Understanding feature gates".

.Prerequisites

> * You have logged into the {oc-first}.

Reviewer 2: you can not log in to oc without the cluster being installed. I think this is more "have oc and installer downloaded".

Author: Updated as per the suggestion.

.Procedure

. Edit the `install-config.yaml` file to define the arbiter node alongside control plane nodes.
+
.Example `install-config.yaml` configuration for deploying an arbiter node
[source,yaml]
----
apiVersion: v1
baseDomain: devcluster.openshift.com
compute:
  - architecture: amd64
    hyperthreading: Enabled
    name: worker
    platform: {}
    replicas: 0
arbiter: <1>
  architecture: amd64
  hyperthreading: Enabled
  replicas: 1 <2>
  name: arbiter <3>
  platform:
    baremetal: {}
controlPlane: <4>
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    baremetal: {}
  replicas: 2 <5>
platform:

>   baremetal:

Author: @eggfoobar 
WDYT?
If we need to include these fields, could you please share the updated example configuration?

>   baremetal:

Reviewer 3: I think linking to the `baremetal` information will be useful, but since everything else is pretty much the same aside from the detail about `role` mentioned below, I think we're fine to not add it here so it's not too much information.

# ...
    hosts:
      - name: cluster-master-0
        role: master
# ...
      - name: cluster-master-1
        role: master
        ...
      - name: cluster-arbiter-0
        role: arbiter
# ...
----
<1> Defines the arbiter machine pool. You must configure this field to deploy a cluster with an arbiter node.

> <2> Set the `replicas` field to `1` or more for the arbiter pool.

Author: Updated as per the suggestion.

<3> Specifies a name for the arbiter machine pool.
<4> Defines the control plane machine pool.
<5> When an arbiter pool is defined, two control plane replicas are valid.

. Save the modified `install-config.yaml` file.

> // TBD - Should we add a link to docs on how users can proceed with cluster installation post performing this procedure?

Author: Updated as per the suggestion.

//
//  *installing/installing_bare_metal/ipi/ipi-install-installation-workflow.adoc

:_mod-docs-content-type: PROCEDURE
[id="ipi-install-config-local-arbiter-node_{context}"]
= Configuring a local arbiter node

You can configure an {product-title} cluster with two control plane nodes and one local arbiter node so to retain high availability (HA) while reducing infrastructure costs for your cluster. This configuration is supported only for bare-metal installations.

:FeatureName: Configuring a local arbiter node
include::snippets/technology-preview.adoc[]

A local arbiter node is a lower-cost, co-located machine that participates in control plane quorum decisions. Unlike a standard control plane node, the arbiter node does not run the full set of control plane services. You can use this configuration to maintain HA in your cluster with only two fully provisioned control plane nodes instead of three.

[IMPORTANT]
====
You can configure a local arbiter node only. Remote arbiter nodes are not supported.
====

To deploy a cluster with two control plane nodes and one local arbiter node, you must define the following nodes in the `install-config.yaml` file:

> * 2 control plane nodes

Reviewer 2: which will also act as worker nodes and run the workload. For the control plane part, 4 threads / 16 GB of RAM and 120 GB of  SSD or equivalent storage is required as minimum. Enough capacity for the workload (incl potential software defined storage solutions) is additionally needed.

Author: Updated as per the suggestion.

For more information about feature gates, see "Understanding feature gates".

The arbiter node must meet the following minimum system requirements:

* 2 threads
* 8 GB of RAM
* 120 GB of SSD or equivalent storage

The arbiter node must be located in a network environment with an end-to-end latency of less than 500 milliseconds, including disk I/O. In high-latency environments, you might need to apply the `etcd` slow profile.

.Prerequisites

* You have downloaded {oc-first} and the installation program.
* You have logged into the {oc-first}.

.Procedure

. Edit the `install-config.yaml` file to define the arbiter node alongside control plane nodes.
+
.Example `install-config.yaml` configuration for deploying an arbiter node
[source,yaml]
----
apiVersion: v1
baseDomain: devcluster.openshift.com
compute:
  - architecture: amd64
    hyperthreading: Enabled
    name: worker
    platform: {}
    replicas: 0
arbiter: <1>
  architecture: amd64
  hyperthreading: Enabled
  replicas: 1 <2>
  name: arbiter <3>
  platform:
    baremetal: {}
controlPlane: <4>
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    baremetal: {}
  replicas: 2 <5>

> platform:

Reviewer 2: I think this is missing the:
featureSet: TechPreviewNoUpgrade
attribute, right? 
@Neilhamza  and @eggfoobar , please confirm!
